sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -322,12 +376,25 @@ class SpillableHostBufferHandle private (
     if (!spillable) {
       0L
     } else {
-      val spilled = synchronized {
-        if (disk.isEmpty && host.isDefined) {
+      val thisThreadSpills = synchronized {
+        if (!closed && disk.isEmpty && host.isDefined && !spilling) {
+          spilling = true
+          // incRefCount here so that if close() is called
```

Comment:
This is maybe not needed anymore, but it's not incorrect either

Reply:
It looks good.